### PR TITLE
fix: datepicker icon color for inverse variant

### PR DESCRIPTION
**Pull request by ghost**
**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

Screen Shot 2022-09-30 at 3 15 54 PM

### CODE REVIEW

File: packages/paste-core/components/input/src/Input.tsx [hunk 1]
```diff
@@ -66,6 +66,7 @@ export const InputElement = React.forwardRef<HTMLInputElement, InputProps>(({ele
       width="100%"
       variant={props.variant}
       ref={ref}
+      colorScheme={props.variant === 'inverse' ? 'dark' : 'normal'}
```
[the remark this answers is not in the document]
Comment (gloriliale) [reply packages/paste-core/components/input/src/Input.tsx:69]: thought: the icon is also the wrong color when the component is in dark mode. Could we maybe make a token that we feed to a colorScheme prop that's 'normal' for default mode and 'dark' for dark mode? We'd then have to override it here for inverse mode, but I do think that the dark mode case is more common.

[the remark this answers is not in the document]
Comment (TheSisb) [reply packages/paste-core/components/input/src/Input.tsx:69]: I have some thoughts on this matter. 

My first thought was to detect which theme was active and to use that information in this conditional, however we don't expose the current active theme from `useTheme` or anything else. If we've made this this far without it, we probably don't want to expose it now. Thinking on it further, exposing that information can fracture our encapsulation which currently gives tokens so much power.

My next thought was akin to Glorilí, in that we could make a couple tokens for this:

`default theme`:
```
color-scheme: 'light'
color-scheme-inverse: 'dark'
```


`dark theme`:
```
color-scheme: 'dark'
color-scheme-inverse: 'light'
```
This way it works for any theme if they're dark or not, and let's people change the values for custom themes which may tend darker or lighter too.


And my third approach was to set the rule on our global css injeciton: 
image
We set the class `paste-theme-provider` in our ThemeProvider and we can set this CSS rule there globally based on the selected theme.

[the remark this answers is not in the document]
Comment (gloriliale) [reply packages/paste-core/components/input/src/Input.tsx:69]: It feels like the token-based approach matches the way our system works the best. We don't have anything else that we set globally based on theme (the third option) do we?

**Comment by SiTaggart** [comment]
Closing in favour of https://github.com/twilio-labs/paste/pull/3130

**Pull request closed without merging**